Update to Polymer 1

Mihaiii opened the pull request:
[41b7847](https://github.com/Juicy/juicy-markdown-editor/pull/3/commits/41b7847784795f8da76f08923691ab9770c925cf) will be necessary.

ErlendLandro commented:
Great that you have done this! When you make a PR, please assign a reviewer. In this case, I would recommend @tomalec 

Remember to close #2 when you're done with this one too.

tomalec commented on juicy-markdown-tabbededitor.html:
```diff
@@ -5,16 +5,10 @@
-
-		<style>
+<link rel="import" href="../juicy-markdown-editor/juicy-markdown-editor.html">
+<head>
```
You don't need head tag

tomalec commented:
Also I'd bump version separately, after the PR is merged, also I'd bump MAJOR one.

Mihaiii commented:
It should be related to the relative paths.
It works like this:
<img width="960" alt="2017-06-14 27" src="https://user-images.githubusercontent.com/8951553/27148573-1085f8be-5141-11e7-9e48-b2e1b73a79d3.png">

warpech commented:
@Mihaiii is this ready for a re-review?

Mihaiii commented:
I didn't really address the main problems @tomalec talked about:

> if I run http://127.0.0.1:8080/components/juicy-markdown-tabbededitor/ console is full of errors and the editor is not show.

IMO they aren't related to the changes I made or, if they are, I don't know what I should change.

warpech commented:
@tomalec can you pls add some hints for @Mihaiii how to proceed?

warpech commented on juicy-markdown-tabbededitor.html:
```diff
@@ -223,56 +215,78 @@
-	      	}    
+	      	},
+			setTabButtonClass: function(idx, oldIdx){
+				var tabHeader = this.$$('.tabHeader');
```
Simpler:

```
var tabButtons = this.querySelectorAll(". tabButton"); 
var oldButton = tabButtons[oldIdx];
var newButton = tabButtons[idx];
```

warpech commented on juicy-markdown-tabbededitor.html:
```diff
@@ -223,56 +215,78 @@
 
 	      		this.index = 2;
+				this.setTabButtonClass(2, this.oldIndex);
 	      	},	
```
Again, the indentation looks mixed up:

<img width="714" alt="screen shot 2017-07-06 at 12 15 07" src="https://user-images.githubusercontent.com/566463/27906666-cd9b9634-6244-11e7-9289-da299e73d81e.png">

warpech commented on juicy-markdown-tabbededitor.html:
```diff
@@ -223,56 +215,78 @@
 	      	buttonClick1: function(event, detail, sender) { 
 
 	      		this.index = 0;
+				this.setTabButtonClass(0, this.oldIndex);
```
Simpler:

```
this.setTabButtonClass(this.index, this.oldIndex);
```

warpech commented on bower.json:
```diff
@@ -25,7 +25,7 @@
   ],
   "dependencies": {
-    "polymer": "polymer/polymer#~0.5.5",
+    "polymer": "polymer/polymer#~1.0.0",
```
This is not a correct selector, because it will match version range `>= 1.0.0` and `< 1.1.0`.

What you mean is `"polymer/polymer#^1.0.0"`. This means `>= 1.0.0`

Mihaiii commented:
@warpech Thank you so much for this! Made the commit with requested changes and it's good for a re-review!

Mihaiii commented:
Update according to StarcounterApps/Content#71 (comment) for separation of concerns. This includes my later change - for not updating content of ace-editor if it's already updated - that comes with commit StarcounterApps/Content@c260048 .

Mihaiii commented:
@warpech Updated "juicy-markdown-tabbededitor.html" according to the changes that were made in Content app (as I wrote in the description). Can I bump major and make a release?

warpech commented:
I have already reviewed it as part of the Content app so I say yes. @tomalec could you please do a 5 minute review?

Mihaiii commented:
Hi @tomalec !
Updated bower.json (now it has `to-markdown`) and also I added some properties in order to pass them further on to `juicy-markdown-editor`.
Related to the image upload, I had a look around and IMO it's not related to these changes. 
For example: 
- I'm not sure, but probably [here](https://github.com/Juicy/juicy-markdown-editor/blob/master/juicy-markdown-editor.html#L53) the default value should be `x-file`.
- I guess [here](https://github.com/Juicy/juicy-markdown-editor/blob/master/test/mockServer.js#L6) we can probably also check for `xhr.xFile` for `xFile` variable because of [this](https://github.com/Juicy/juicy-filedrop/blob/master/juicy-filedrop.html#L139) line.
- I couldn't really understood what [this](https://github.com/Juicy/juicy-filedrop/blob/master/juicy-filedrop.html#L207) is or what it should do.
- We could probably check for `this` [here](https://github.com/Juicy/juicy-filedrop/blob/master/juicy-filedrop.html#L130).

All in all, I would remove the file drop option at this moment and restore it when we have more time. Would that be OK?

tomalec commented:
Why do we need `to-markdown`? what it is? previously it was not needed, as markdown transformation was  done by the `juicy-markdown-editor` (via Marked AFAIK).

> All in all, I would remove the file drop option at this moment and restore it when we have more time. Would that be OK?

I would't, @warpech WDYT?

Mihaiii commented:
Wanted to update README.md, but I guess we have to first see if we'll implement the changes to `juicy-markdown-editor` and `juicy-filedrop` for the "file drop" option. So I'll just write here: `to-markdown` is an HTML to Markdown converter. We need it in case the users of `juicy-markdown-tabbededitor` want to add more tabs with more editors (for example `juicy-ace-editor`), in which case we need `to-markdown` in order to provide the input for `juicy-markdown-editor`. A working example can be found in Content app on develop branch ( https://github.com/StarcounterApps/Content ).

What I wanted to say with the previous comment was that the "file drop" bug has nothing to do with `juicy-markdown-tabbededitor` imo; it's related to `juicy-filedrop` and maybe `juicy-markdown-editor`. Shouldn't any issues about "file drop" be raised there and solved there, not here? Since `juicy-filedrop` seems deprecated , can't we remove this option (file drop) from `juicy-markdown-tabbededitor` until a fix is made there (after which we'll update `juicy-markdown-tabbededitor` again)?

Mihaiii commented:
How should I proceed?
`mockServer.js` and `sinon.js` have been removed from `index.html` here
https://github.com/Juicy/juicy-markdown-tabbededitor/pull/2/commits/b467b0c88366edd97597b9df1c50c3447e1fcb50#diff-eacf331f0ffc35d4b482f1d15a887d3b .
I have noticed this comment: https://github.com/Juicy/juicy-filedrop/commit/470dd616021f9e732c28386fe8379eba49e22826#diff-eacf331f0ffc35d4b482f1d15a887d3bR34

warpech commented:
Since this element is not used in any Starcounter app, I think we should pause this PR for the time being. Please unpause if someone wants to use it.

warpech commented:
Closing as won't merge. We have decided to deprecate this repo. See the deprecation announcement at #6.

Thanks @Mihaiii so much for your involvement.

It is fine to continue using a fork of this element in the Contents app.
